File: frontend/src/types/shop.ts
export type ShopItemType = "THEME" | "AVATAR_FRAME" | "BADGE" | "TITLE" | "COSMETIC";

export interface ShopItem {
  id: string;
  code: string;
  name: string;
  description: string;
  item_type: ShopItemType;
  cost_gold: number;
  asset_key: string;
  is_active: boolean;
  created_at: string;
}

export interface InventoryItem {
  id: string;
  character_id: string;
  shop_item_id: string;
  is_equipped: boolean;
  acquired_at: string;
  shop_item: ShopItem;
}

export interface PurchaseResponse {
  inventory_item: InventoryItem;
  remaining_gold: number;
  detail: string;
}

File: backend/app/schemas/shop.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict
from app.models.enums import ShopItemType


class ShopItemOut(BaseModel):
    id: uuid.UUID
    code: str
    name: str
    description: str
    item_type: ShopItemType
    cost_gold: int
    asset_key: str
    is_active: bool
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class InventoryItemOut(BaseModel):
    id: uuid.UUID
    character_id: uuid.UUID
    shop_item_id: uuid.UUID
    is_equipped: bool
    acquired_at: datetime
    shop_item: ShopItemOut

    model_config = ConfigDict(from_attributes=True)


class PurchaseResponse(BaseModel):
    inventory_item: InventoryItemOut
    previous_gold: int
    gold_spent: int
    remaining_gold: int
    newly_owned: bool = True
    already_owned: bool = False
    detail: str = "Item purchased successfully."
